src/main/java/com/github/igotyou/FactoryMod/recipes/PrintingPlateJsonRecipe.java

```diff
@@ -0,0 +1,348 @@
+package com.github.igotyou.FactoryMod.recipes;
+
+import com.github.igotyou.FactoryMod.FactoryMod;
+import com.github.igotyou.FactoryMod.factories.FurnCraftChestFactory;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import net.minecraft.server.v1_16_R1.NBTTagCompound;
+import net.minecraft.server.v1_16_R1.NBTTagList;
+import net.minecraft.server.v1_16_R1.NBTTagString;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BookMeta;
+import org.bukkit.inventory.meta.ItemMeta;
+import vg.civcraft.mc.civmodcore.api.ItemAPI;
+import vg.civcraft.mc.civmodcore.itemHandling.ItemMap;
+
+import java.util.Arrays;
+import java.util.UUID;
+
+public class PrintingPlateJsonRecipe extends PrintingPlateRecipe {
+	/*
+	 * WARNING
+	 *
+	 * This inherits from PrintingPlateRecipe
+	 *                            ^^^^^
+	 * Not PrintingPressRecipe, as would normally be expected.
+	 *             ^^^^^
+	 *
+	 * This is because this recipe creates a printing plate, like the super class.
+	 */
+
+	public PrintingPlateJsonRecipe(String identifier, String name, int productionTime, ItemMap input, ItemMap output) {
+		super(identifier, name, productionTime, input, output);
+	}
+
+	@Override
+	public boolean enoughMaterialAvailable(Inventory i) {
+		String[] pages = String.join("", ((BookMeta) getBook(i).getItemMeta()).getPages()).split("<<PAGE>>");
+
+		for (String page : pages) {
+			page = page.replace("\\", "\\\\");
+			try {
+				Gson gson = new Gson();
+				JsonElement element = gson.fromJson(page, JsonElement.class);
+
+				String result = checkForIllegalSections(element);
+				if (result != null) {
+					factioryError(i, "Banned Tag Error", "Error Message: " + result);
+
+					return false;
+				}
+			} catch (JsonSyntaxException e) {
+				factioryError(i, "JSON Syntax Error", e.toString());
+
+				return false;
+			} catch (StackOverflowError e) {
```

Comment:
I'd prefer having a max nesting depth of like 10-15, having to catch StackOverflowErrors as input validation is pretty ugly

Reply:
I'll lower the limit. I'd prefer to catch it just in case though. I overload the exception to also tell the function the nesting depth has been reached.

Reply:
Oh or maybe I should just return as a syntax error if the nesting limit has been reached. I'll do that instead.